commit b146fb0370d2d9d93a96b25139f88ec11a338e89
Author: Steve Muckle <smuckle@google.com>
Date:   Mon Dec 9 16:00:11 2019 -0800

    eas: sched_boost: do not require optional tunables
    
    These tunables may not be present in all implementations.
    
    Also return TCONF if schedtune is not found at the expected
    Android mountpoint.
    
    Bug: 144809570
    Change-Id: I18b8711dca0ae5d4d7be58981a445bcc488480cc
    Signed-off-by: Steve Muckle <smuckle@google.com>

diff --git a/testcases/kernel/sched/eas/sched_boost.c b/testcases/kernel/sched/eas/sched_boost.c
index 1f65b65e6..84d17084c 100644
--- a/testcases/kernel/sched/eas/sched_boost.c
+++ b/testcases/kernel/sched/eas/sched_boost.c
@@ -139,6 +139,9 @@ static void run(void)
 	sem_init(&test_sem, 0, 0);
 	sem_init(&result_sem, 0, 0);
 
+	if (access("/dev/stune", R_OK))
+		tst_brk(TCONF, "schedtune not detected");
+
 	test_cpu = tst_ncpus() - 1;
 	printf("Running %ld tests for %d sec\n", NUM_TESTS,
 	       TEST_LENGTH_USEC / USEC_PER_SEC);
@@ -149,14 +152,24 @@ static void run(void)
 	 * kernel/sched/tune.c needs to be increased.
 	 */
 	SAFE_MKDIR(STUNE_TEST_PATH, 0777);
-	SAFE_FILE_PRINTF(STUNE_TEST_PATH "/schedtune.colocate",
-			 "0");
-	SAFE_FILE_PRINTF(STUNE_TEST_PATH "/schedtune.prefer_idle",
-			 "0");
-	SAFE_FILE_PRINTF(STUNE_TEST_PATH "/schedtune.sched_boost_enabled",
-			 "1");
-	SAFE_FILE_PRINTF(STUNE_TEST_PATH "/schedtune.sched_boost_no_override",
-			 "0");
+	if (access(STUNE_TEST_PATH "/schedtune.colocate", W_OK) != -1) {
+		SAFE_FILE_PRINTF(STUNE_TEST_PATH "/schedtune.colocate",
+				 "0");
+	}
+	if (access(STUNE_TEST_PATH "/schedtune.prefer_idle", W_OK) != -1) {
+		SAFE_FILE_PRINTF(STUNE_TEST_PATH "/schedtune.prefer_idle",
+				 "0");
+	}
+	if (access(STUNE_TEST_PATH "/schedtune.sched_boost_enabled", W_OK) !=
+	    -1) {
+		SAFE_FILE_PRINTF(STUNE_TEST_PATH "/schedtune.sched_boost_enabled",
+				 "1");
+	}
+	if (access(STUNE_TEST_PATH "/schedtune.sched_boost_no_override",
+		   W_OK) != -1) {
+		SAFE_FILE_PRINTF(STUNE_TEST_PATH "/schedtune.sched_boost_no_override",
+				 "0");
+	}
 
 	SAFE_PTHREAD_CREATE(&test_thread, NULL, test_fn, NULL);
 
